git-path/src/realpath.rs

```diff
@@ -0,0 +1,84 @@
+/// the error returned by [`realpath()`][super::realpath()].
+#[derive(Debug, thiserror::Error)]
+#[allow(missing_docs)]
+pub enum Error {
+    #[error("The maximum allowed number {} of symlinks in path is exceeded", .max_symlinks)]
+    MaxSymlinksExceeded { max_symlinks: u8 },
+    #[error(transparent)]
+    ReadLink(#[from] std::io::Error),
+    #[error("Empty is not a valid path")]
+    EmptyPath,
+    #[error("Parent component of {} does not exist", .path)]
+    MissingParent { path: String },
+}
+
+pub(crate) mod function {
```

Comment:
Now that it's own module, what is the benefit of nesting it in another module? Or why is it not just `pub fn realpath()`

Reply:
Having `fn realpath()`  in `pub mod realpath` would make its path (also in the docs)` realpath::realpath` even though one might import it later to `crate::realpath()`, which leaves it at two places.

The package private module `function` is just there to keep `fn realpath()` in `mod realpath` but hide it from being exported there. Instead it's being pulled up later to be exported in `crate::realpath()` while it's actually located in `mod realpath`. There, however, it's not available so to the API user it exists only in one place, not in two. Ambiguous paths I avoid and they can only happen with `prelude` modules which are designed to contain typical imports. The latter you find only in `git-repository` I think.